packages/contracts/src/interest/MultiTokenVault.sol

```diff
@@ -1,232 +1,280 @@
     {
-        return previewRedeemForDepositPeriod(shares, depositPeriod, currentPeriodElapsed);
+        return convertToAssetsForDepositPeriod(shares, depositPeriod, redeemPeriod);
     }
 
     /**
-     * @dev See {IMultiTokenVault-redeemForDepositPeriod}
+     * @inheritdoc IMultiTokenVault
      */
-    function redeemForDepositPeriod(uint256 shares, address receiver, address owner, uint256 depositPeriod)
+    function previewRedeemForDepositPeriod(uint256 shares, uint256 depositPeriod)
         public
+        view
+        virtual
         returns (uint256 assets)
     {
-        return redeemForDepositPeriod(shares, receiver, owner, depositPeriod, currentPeriodElapsed);
+        return previewRedeemForDepositPeriod(shares, depositPeriod, currentTimePeriodsElapsed());
     }
 
-    // =============== ERC4626 and ERC20 ===============
-
     /**
-     * @dev See {IMultiTokenVault-asset}
+     * @inheritdoc IMultiTokenVault
      */
-    function asset() public view virtual override returns (address asset_) {
-        return address(ASSET);
+    function currentTimePeriodsElapsed() public view virtual returns (uint256) {
+        return _currentTimePeriodsElapsed;
     }
 
     /**
-     * @dev See {ERC20-decimals}
+     * @dev Returns true if this contract implements the interface defined by `interfaceId`.
+     *      This function checks for support of the IERC1155 interface, IMultiTokenVault interface,
+     *      and delegates to the super class for any other interface support checks.
+     *
+     * @param interfaceId The identifier of the interface to check for support.
+     *
+     * @return bool True if the contract supports the requested interface.
      */
-    function decimals() public view virtual returns (uint8) {
-        return IERC20Metadata(address(ASSET)).decimals();
+    function supportsInterface(bytes4 interfaceId) public view virtual override(IERC165, ERC1155) returns (bool) {
+        return interfaceId == type(IMultiTokenVault).interfaceId || super.supportsInterface(interfaceId);
     }
 
-    // =============== Utility ===============
-
     /**
-     * @dev See {IMultiTokenVault-getCurrentTimePeriodsElapsed}
+     * @dev An internal function to implement the functionality of depositing assets into the vault
+     *      and mints shares for the current time period.
+     *
+     * @param assets The amount of the ERC-20 underlying assets to be deposited into the vault.
+     * @param receiver The address that will receive the minted shares.
+     * @param caller The address of who is depositing the assets.
+     * @param depositPeriod The time period in which the assets are deposited.
+     * @param shares The amount of ERC-1155 tokens minted.
      */
-    function currentTimePeriodsElapsed() public view virtual returns (uint256 currentTimePeriodsElapsed_) {
-        return currentPeriodElapsed;
+    function _deposit(uint256 assets, address receiver, address caller, uint256 depositPeriod, uint256 shares)
```

Comment:
i'd follow the ERC4626 ordering here.  note especially that caller is before receiver.

    function _deposit(address caller, address receiver, uint256 assets, uint256 shares) internal virtual {